rust/src/router.rs

```diff
@@ -533,8 +542,8 @@ impl Router {
     /// * See also how to [filter these RTP capabilities](https://mediasoup.org/documentation/v3/tricks/#rtp-capabilities-filtering)
     ///   before using them into a client.
     #[must_use]
-    pub fn rtp_capabilities(&self) -> &RtpCapabilitiesFinalized {
-        &self.inner.rtp_capabilities
+    pub fn rtp_capabilities(&self) -> RtpCapabilitiesFinalized {
+        self.inner.rtp_capabilities.lock().unwrap().clone()
```

Comment:
I don't understand why I cannot have a mutable internal property and a method that returns its reference. Or maybe I'm wrong and it's possible.

Reply:
Note that the downside is the extra clone() in the getter.

Reply:
Of course. Well, the API has changed already due to return of an owned struct, but it is not really avoidable here.